entry_types/scrolled/lib/pageflow_scrolled.rb

```diff
@@ -7,10 +7,11 @@ def plugin
       PageflowScrolled::Plugin.new
     end
 
-    def entry_type
+    def entry_type(config)
       Pageflow::EntryType.new(name: 'scrolled',
                               frontend_app: PageflowScrolled::EntriesController.action(:show),
-                              editor_fragment_renderer: editor_fragment_renderer)
+                              editor_fragment_renderer: editor_fragment_renderer,
+                              configuration: Plugin::ScrolledConfiguration.new(config))
```

Comment:
I think a lot less changes like this would be necessary if we decided to pass a configuration class instead of a configuration object to `EntryType.new`. We could require these classes to have an initializer which takes a `config` as its only parameter. `for_entry_type` below could then instantiate that class and keep it in local hash which maps entry type names to entry type configuration objects the way it was in a previous iteration of this PR.